org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/semantictokens/SemanticTokensService.java

```diff
@@ -105,20 +135,20 @@ public SemanticTokens semanticTokensFull(final Document document, final XtextRes
 				Integer positionTokenType = getTokenType(lightweightPosition.getIds());
 				Integer positionTokenModifiers = getTokenModifiers(lightweightPosition.getIds());
 
-				if (positionTokenType != 0 || positionTokenModifiers != 0) {
-					int deltaOffset = lightweightPosition.getOffset() - lastOffset;
-					reader.skip(deltaOffset);
-					Position position = reader.getPosition();
-					int deltaLine = position.getLine() - lastLine;
-					data.add(deltaLine); // delta line
-					lastLine = position.getLine();
-					// delta start relative to previous token if on the same line or to 0
-					data.add(deltaLine == 0 ? deltaOffset : position.getCharacter());
-					lastOffset = lightweightPosition.getOffset();
-					data.add(lightweightPosition.getLength()); // length
-					data.add(positionTokenType); // token type
-					data.add(positionTokenModifiers); // token modifiers
-				}
+				int deltaOffset = lightweightPosition.getOffset() - lastOffset;
+				reader.skip(deltaOffset);
+				Position position = reader.getPosition();
+				
+				int deltaLine = position.getLine() - lastLine;
+				lastLine = position.getLine();
+				lastOffset = lightweightPosition.getOffset();
+				
+				data.add(deltaLine); // delta line
+				// delta start relative to previous token if on the same line or to 0
+				data.add(deltaLine == 0 ? deltaOffset : position.getCharacter());
+				data.add(lightweightPosition.getLength()); // length
+				data.add(positionTokenType); // token type
```

Comment:
I think here is where we missed a `positionTokenType - 1` to compensate for the chosen communication channel `0 == not contained in list`

Reply:
to use 0 as an indicator for not supported tokenTypes we would need to add a placeholder tokenType in the list of supported types. Otherwise we would not be able to use the first token Type in the list of supported types. That's why I changed the indicator to -1